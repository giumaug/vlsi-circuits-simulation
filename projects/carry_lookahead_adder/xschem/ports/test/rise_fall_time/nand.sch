v {xschem version=3.1.0 file_version=1.2
}
G {}
K {}
V {}
S {}
E {}
N -0 200 50 200 {
lab=GND}
N 0 50 70 50 {
lab=#net1}
N 50 200 180 200 {
lab=GND}
N 180 170 180 200 {
lab=GND}
N 70 50 180 50 {
lab=#net1}
N 180 50 180 70 {
lab=#net1}
N 50 140 120 140 {
lab=#net2}
N -0 110 -0 200 {
lab=GND}
N 120 50 120 100 {
lab=#net1}
C {devices/vsource.sym} 0 80 0 0 {name=V1 value=1.8}
C {devices/vsource.sym} 50 170 0 0 {name=V2 value="pwl 0ps 0 100ps 0 150ps 1.8 500ps 1.8 550ps 0 1ns 0"}
C {devices/gnd.sym} 180 200 0 0 {name=l1 lab=GND}
C {devices/code_shown.sym} 500 240 0 0 {name=s1 only_toplevel=false value=".lib /usr/local/share/sky130_fd_pr/models/sky130.lib.spice tt
.tran 20p 1n
.save all"}
C {devices/code_shown.sym} 480 40 0 0 {name=tpdr only_toplevel=false value=".measure tran tpdr TRIG v(net2) VAL=0.9 FALL=1 TARG v(net3) VAL=0.9 RISE=1"}
C {devices/code_shown.sym} 480 150 0 0 {name=tpdf  only_toplevel=false value=".measure tran tpdf TRIG v(net2) VAL=0.9 RISE=1 TARG v(net3) VAL=0.9 FALL=1"}
C {/home/peppe/Scrivania/vlsi/circuits-simulation/projects/carry_lookahead_adder/xschem/ports/nand.sym} 180 120 0 0 {name=x1}
