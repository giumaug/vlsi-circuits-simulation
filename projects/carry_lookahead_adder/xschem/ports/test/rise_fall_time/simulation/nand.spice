** sch_path:
*+ /home/peppe/Scrivania/vlsi/circuit-simulation/projects/carry_lookahead_adder/xschem/ports/test/rise_fall_time/nand.sch
**.subckt nand
V1 net1 GND 1.8
.save i(v1)
V2 net2 GND pwl 0ps 0 100ps 0 150ps 1.8 500ps 1.8 550ps 0 1ns 0
.save i(v2)
x1 net1 net2 net3 net1 GND nand
**** begin user architecture code

.lib /usr/local/share/sky130_fd_pr/models/sky130.lib.spice tt
.tran 20p 1n
.save all


.measure tran tpdr TRIG v(net2) VAL=0.9 FALL=1 TARG v(net3) VAL=0.9 RISE=1


.measure tran tpdf TRIG v(net2) VAL=0.9 RISE=1 TARG v(net3) VAL=0.9 FALL=1

**** end user architecture code
**.ends

* expanding   symbol:  nand.sym # of pins=5
** sym_path:
*+ /home/peppe/Scrivania/vlsi/circuit-simulation/projects/carry_lookahead_adder/xschem/ports/nand.sym
** sch_path:
*+ /home/peppe/Scrivania/vlsi/circuit-simulation/projects/carry_lookahead_adder/xschem/ports/nand.sch
.subckt nand in1 in2 out vdd vss
*.ipin in1
*.ipin in2
*.ipin vdd
*.ipin vss
*.opin out
XM1 net1 in2 vss vss sky130_fd_pr__nfet_01v8 L=0.15 W=2 nf=1 ad='int((nf+1)/2) * W/nf * 0.29' as='int((nf+2)/2) * W/nf * 0.29'
+ pd='2*int((nf+1)/2) * (W/nf + 0.29)' ps='2*int((nf+2)/2) * (W/nf + 0.29)' nrd='0.29 / W' nrs='0.29 / W'
+ sa=0 sb=0 sd=0 mult=1 m=1
XM2 out in1 net1 net1 sky130_fd_pr__nfet_01v8 L=0.15 W=2 nf=1 ad='int((nf+1)/2) * W/nf * 0.29' as='int((nf+2)/2) * W/nf * 0.29'
+ pd='2*int((nf+1)/2) * (W/nf + 0.29)' ps='2*int((nf+2)/2) * (W/nf + 0.29)' nrd='0.29 / W' nrs='0.29 / W'
+ sa=0 sb=0 sd=0 mult=1 m=1
XM6 out in1 vdd vdd sky130_fd_pr__pfet_01v8 L=0.15 W=3.8 nf=1 ad='int((nf+1)/2) * W/nf * 0.29' as='int((nf+2)/2) * W/nf * 0.29'
+ pd='2*int((nf+1)/2) * (W/nf + 0.29)' ps='2*int((nf+2)/2) * (W/nf + 0.29)' nrd='0.29 / W' nrs='0.29 / W'
+ sa=0 sb=0 sd=0 mult=1 m=1
XM3 out in2 vdd vdd sky130_fd_pr__pfet_01v8 L=0.15 W=3.8 nf=1 ad='int((nf+1)/2) * W/nf * 0.29' as='int((nf+2)/2) * W/nf * 0.29'
+ pd='2*int((nf+1)/2) * (W/nf + 0.29)' ps='2*int((nf+2)/2) * (W/nf + 0.29)' nrd='0.29 / W' nrs='0.29 / W'
+ sa=0 sb=0 sd=0 mult=1 m=1
.ends

.GLOBAL GND
.end
