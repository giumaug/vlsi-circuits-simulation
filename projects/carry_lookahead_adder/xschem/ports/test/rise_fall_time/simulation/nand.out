
Note: No compatibility mode selected!


Circuit: ** sch_path:

binary raw file "/home/peppe/Scrivania/vlsi/circuit-simulation/projects/carry_lookahead_adder/xschem/ports/test/rise_fall_time/simulation/nand.raw"
Doing analysis at TEMP = 27.000000 and TNOM = 27.000000

Note: include: has no value, DC 0 assumed
No. of Data Columns : 21  

Initial Transient Solution
--------------------------

Node                                   Voltage
----                                   -------
net1                                       1.8
net2                                         0
x1.net1                                1.66415
net3                                       1.8
sky130_fd_pr__esd_nfet_05v0_nvt.pm3               0
                                             0
v2#branch                                    0
v1#branch                         -2.10579e-12


No. of Data Rows : 71

No .measure possible in batch mode (-b) with -r rawfile set!
Remove rawfile and use .print or .plot or
select interactive mode (optionally with .control section) instead.


Total analysis time (seconds) = 0.001

Total elapsed time (seconds) = 5.326 

Total DRAM available = 15650.727 MB.
DRAM currently available = 10922.277 MB.
Maximum ngspice program size =  352.148 MB.
Current ngspice program size =  333.074 MB.

Shared ngspice pages =   11.488 MB.
Text (code) pages =    5.297 MB.
Stack = 0 bytes.
Library pages =  330.250 MB.

